apps/jubmoji-quest/src/hooks/useLeaderboard.ts

```diff
@@ -0,0 +1,81 @@
+import { getClientPathToCircuits } from "@/lib/config";
+import { createJubmojiQuestProof } from "@/lib/proving";
+import { JubmojiQuest } from "@/types";
+import { Prisma } from "@prisma/client";
+import { Jubmoji } from "jubmoji-api";
+import { useMutation, useQuery } from "react-query";
+
+type UpdateLeaderboardMutationProps = {
+  jubmojis?: Jubmoji[];
+  quest?: JubmojiQuest | null;
+};
+
+// Returns the leaderboard for a given quest
+export const useGetLeaderBoard = (questId: string | number) => {
+  return useQuery({
+    queryKey: ["leaderboard", questId],
+    queryFn: async (): Promise<
+      Record<number, string | number> | { error: string }
+    > => {
+      // After successful update, re-fetch the leaderboard
+      const url = new URL("/api/team-leaderboard", window.location.origin);
+      url.searchParams.append("questId", `${questId}`);
+      const leaderboardResponse = await fetch(url, {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+
+      if (!leaderboardResponse.ok) {
+        return { error: "Could not fetch new leaderboard" };
+      }
+
+      const { scoreMap } = await leaderboardResponse.json();
+      return scoreMap;
+    },
+  });
+};
+
+// Mutation for updating the leaderboard
+export const useUpdateLeaderBoardMutation = () => {
+  return useMutation({
+    mutationKey: "updateLeaderboard",
+    mutationFn: async ({ jubmojis, quest }: UpdateLeaderboardMutationProps) => {
+      if (!jubmojis) {
+        throw new Error("You must have jubmojis to participate in this quest!");
```

Comment:
Ideally we have some unified handling across the entire app when proofs are being made, but for now can we catch this and just show an alert?

Reply:
this is handled here https://github.com/jubmoji/jubmoji.quest/pull/84/files, when the mutation fails the error message will show the alert